contracts/governance/FeeSharingProxy.sol

```diff
@@ -280,6 +296,17 @@ contract FeeSharingProxy is SafeMath96, IFeeSharingProxy {
 		return (amount, end);
 	}
 
+	function _getTotalWeightedStake(
+		address account,
+		uint256 blockNumber,
+		uint256 timestamp
+	) internal view returns (uint96 weightedStake) {
+		uint96 vestingWeightedStake = staking.getPriorVestingWeightedStake(account, blockNumber, timestamp);
+		weightedStake = staking.getPriorWeightedStake(account, blockNumber, timestamp);
+
+		weightedStake = weightedStake - vestingWeightedStake;
+	}
+
```

Comment:
a) the name is misleading because totalWeightedStake would be considered the total on the system, whereas here it just relates to a user
b) a user can't have vesting + regular stake on his wallet (vesting stake belongs to the vesting contract). therefore, i believe this is unnecessary. when checking for the fees it makes more sense to check if the user is a vesting contract or not.
c) This differntiation would have been required for the total weighted stake, not the user, but above the total weighted stake remains unchanged. this leads to unclaimable fees on the contract